packages/apollo-datasource/src/index.ts

```diff
@@ -6,5 +6,6 @@ export interface DataSourceConfig<TContext> {
 }
 
 export abstract class DataSource<TContext = any> {
+  context!: TContext;
```

Comment:
Would we want this to be `!`'d?  Seems plausible that current implementations don't have it or might not have it?  I think it definitely makes sense to have it non-null asserted in `RESTDataSource` where we can be certain that we'll call it, but this seems a bit different.

Granted, it's just a typing change, so not a runtime disaster, but just curious what the thinking was here.